lib/RGBot.js

```diff
@@ -463,87 +935,130 @@ const RGBot = class {
      * @param {boolean} [options.partialMatch=false] Find blocks whose name / displayName contains blockType. (Ex. 'log' may find any of 'spruce_log', 'oak_log', etc.)
      * @param {boolean} [options.onlyFindTopBlocks=false] Will not return any blocks that are beneath another block
      * @param {number} [options.maxDistance=50] Find any Blocks matching the search criteria up to and including this distance from the Bot
      * @param {boolean} [options.skipClosest=false] Will attempt to locate the next-closest Block. This can be used to skip the closest Block when the Bot encounters an issue collecting it
      * @returns {Block | null}
+     *
+     * @deprecated Since 1.2.0, please use findBlocks(options, blockValueFunction, sortValueFunction)
      */
     findBlock(blockType, options = {}) {
+        this.#logWarn(`findBlock(blockType, options): Deprecated since 1.2.0, please use findBlocks(options, blockValueFunction, sortValueFunction)`)
+        options['blockNames'] = blockType && [blockType] || [];
+        return this.findBlocks(options).shift()?.result
+    }
+
+    /**
+     * Returns the best block that is diggable within a maximum distance from the Bot.
+     * @param {object} [options] - optional parameters
+     * @param {string[]} [options.blockNames=[]] List of blockNames to consider
+     * @param {boolean} [options.partialMatch=false] Consider blocks whose name partially matches one of the blockNames
+     * @param {boolean} [options.onlyFindTopBlocks=false] Only find blocks that don't have a block above them.
+     * @param {number} [options.maxDistance=50] Max range to consider
+     * @param {number} [options.maxCount=1] Max count of matching blocks
+     * @param {function(blockName:string):number} [blockValueFunction] - function to call to get the value of a block based on its name (blockName); should return the best block with the lowest sorting value. A good example function is { return scoreValueOf[blockName] }, where scoreValueOf is the point value or intrinsic value of the block in the game mode being played.  If you don't want a block considered, return a value < 0 for its value.
+     * @param {function(pointValue:number,distance:number,digTime:number):number} [sortValueFunction] - function to call to sort the evaluation of results based on arguments (pointValue, distance, digTime[seconds]); should return the best item with the lowest sorting value.
+     * @returns {FindResult[]} - the best blocks found
+     *
+     * By default, this only sorts results by distance, and assigns all blocks a value of 0.  A good example function is {return -1 * (pointValue - distance - digTime)}
+     *
+     * To get the 'best' block result, call findBlocks(...).shift().  Note that the result may be null|undefined if no blocks were found
+     */
+    findBlocks(options = {}, blockValueFunction = (blockName) => {return 0}, sortValueFunction = (pointValue, distance, digTime) => {return distance}) {
```

Comment:
Similar comment to previous, there shouldn't be any parameters after `options`, and these parameters should be in the options.

Any reason why blockName was added to options? Don't I almost always want to search for a blockName, so it's not optional. The guideline we set (forget where we wrote this down) is that if it is a requirement for the function, or is used in most use cases, it should be a non-optional parameter.

Reply:
I found in practice that there are 2 ways to look for blocks.  
1. Is you know the name(s) of blocks you want, and if they aren't there, move on to the next decision.
2. Is that you just want the block in range that is most valuable.  Thus, blockNames being optional.

Reply:
Ahhh I see. Do we think it might be a better user experience then to have two functions?

findBlocks(blockName, options)
findBlocksSortedByScore(sortFunction, options)

I would just like to avoid new players having to always write `findBlocks({blockName: 'dirt'})` instead of `findBlocks('dirt')`